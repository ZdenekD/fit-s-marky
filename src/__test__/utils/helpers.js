import React from 'react';
import {shallow} from 'enzyme';

/**
 * Factory function to create a shallow wrapper for App component
 * @function setup
 * @param {ReactComponent} Component - React component to render with shallow
 * @param {Object} props - Component props specific to this setup
 * @returns {ShallowWrapper}
 */
export const factory = (Component, props = {}) => {
    const wrapper = shallow(<Component {...props} />);

    return wrapper;
};

/**
 * Return ShallowWrapper containg node(s) with the data-test value
 * @param {ShallowWrapper} wrapper - Enzyme shallow wrapper to search within
 * @param {string} value - data attribute value to find
 * @returns {ShallowWrapper}
 */
export const findComponent = (wrapper, value) => wrapper.find(`[data-test="${value}"]`);
