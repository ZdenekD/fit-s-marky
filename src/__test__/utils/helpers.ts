import {ShallowWrapper} from 'enzyme';

const findComponent = (wrapper: ShallowWrapper, value: string): ShallowWrapper => wrapper.find(`[data-test="${value}"]`);

export default findComponent;
