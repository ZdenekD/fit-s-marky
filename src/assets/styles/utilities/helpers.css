:global ._pull-right {
    float: right !important;
}

:global ._pull-left {
    float: left !important;
}

:global ._italic {
    font-style: italic;
}
