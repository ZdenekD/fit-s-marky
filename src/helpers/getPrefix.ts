const getPrefix = (): string => '_5innv';

export default getPrefix;
