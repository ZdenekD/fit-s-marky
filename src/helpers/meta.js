import React from 'react';

const getMeta = item => (
    [
        <meta name="description" content={item.description} key={item.id} />,
        <meta property="og:title" content={item.title} key={item.id} />,
        <meta property="og:type" content="website" key={item.id} />,
        <meta property="og:description" content={item.description} key={item.id} />,
        <meta property="og:url" content="https://fit-s-marky.cz/" key={item.id} />,
        <meta name="twitter:card" content="summary" key={item.id} />,
        <meta name="twitter:title" content={item.title} key={item.id} />,
        <meta name="twitter:description" content={item.description} key={item.id} />,
        <meta name="twitter:domain" content="https://fit-s-marky.cz/" key={item.id} />,
    ]
);

export default getMeta;
