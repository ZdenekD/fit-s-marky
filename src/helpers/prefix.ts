const prefix = '_5innv';

export default prefix;
