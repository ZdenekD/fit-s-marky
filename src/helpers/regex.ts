interface IRegex {
    text: RegExp
    zip: RegExp
    email: RegExp
}

const regex: IRegex = {
    text: /^[éěëščřžýáäíïťďňöúůüÿÉĚËŠČŘŽÝÁÄÍÏŤĎŇÖÚŮÜŸ \-,.?!'"\w\s\d]+$/,
    zip: /^\d{3}(\s)?\d{2}$/,
    email: /^("([ !#-[\]-~]|\\[ -~])+"|[-a-z0-9!#$%&'*+/=?^_`{|}~]+(\.[-a-z0-9!#$%&'*+/=?^_`{|}~]+)*)@([0-9a-z\u00C0-\u02FF\u0370-\u1EFF]([-0-9a-z\u00C0-\u02FF\u0370-\u1EFF]{0,61}[0-9a-z\u00C0-\u02FF\u0370-\u1EFF])?\.)+[a-z\u00C0-\u02FF\u0370-\u1EFF]([-0-9a-z\u00C0-\u02FF\u0370-\u1EFF]{0,17}[a-z\u00C0-\u02FF\u0370-\u1EFF])?$/i,
};

export default regex;
