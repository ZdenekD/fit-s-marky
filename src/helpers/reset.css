a,
abbr,
address,
article,
aside,
audio,
b,
blockquote,
body,
button,
canvas,
caption,
cite,
code,
dd,
del,
details,
dfn,
div,
dl,
dt,
em,
embed,
fieldset,
figcaption,
figure,
footer,
form,
h1,
h2,
h3,
h4,
h5,
h6,
header,
html,
i,
iframe,
img,
input,
ins,
kbd,
label,
legend,
li,
main,
mark,
nav,
object,
ol,
output,
p,
pre,
q,
ruby,
s,
samp,
section,
select,
small,
span,
strong,
sub,
summary,
sup,
table,
tbody,
td,
textarea,
tfoot,
th,
thead,
time,
tr,
u,
ul,
var,
video {
    margin: 0;
    padding: 0;
    vertical-align: baseline;
    background: transparent;
    font: inherit;
    font-size: 100%;
    outline: 0;
    border: none;
}

*,
*::after,
*::before {
    box-sizing: border-box;
}

body {
    height: 100%;
    overflow-y: scroll;
    line-height: 1;
}

article,
aside,
details,
figcaption,
figure,
footer,
header,
main,
nav,
section {
    display: block;
}

audio,
canvas,
video {
    display: inline-block;
}

ol,
ul {
    list-style: none;
}
