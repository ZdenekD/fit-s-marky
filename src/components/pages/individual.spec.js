import {factory} from '../../__test__/utils/helpers';
import Individual from './individual';

describe('Individual page', () => {
    it('match snapshot', () => {
        const wrapper = factory(Individual);

        expect(wrapper.html()).toMatchSnapshot();
    });
});
