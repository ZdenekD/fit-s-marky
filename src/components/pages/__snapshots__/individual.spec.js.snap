// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Individual page match snapshot 1`] = `
<header>
  <h2>
    Individuální lekce pilates
  </h2>
</header>
<h3>
  Chci instruktora jen pro sebe = soukromá lekce v kanceláři nebo u Vás doma
</h3>
<ul>
  <li>
    Chcete se nejdříve správně naučit pozice a cviky, než půjdete do skupinové lekce?
  </li>
  <li>
    Chcete cvičit pilates, ale nevyhovují Vám časy studiových hodin ve fitness centrech?
  </li>
  <li>
    Nechcete cvičit ve skupinové lekci?
  </li>
  <li>
    Chcete, aby Vám byly lekce Pilates přizpůsobeny na míru?
  </li>
  <li>
    Máte zdravotní problémy: skoliózu, nadváhu, trpíte inkontinencí …?
  </li>
  <li>
    Chcete provádět cviky správně tak, aby mělo cvičení maximální účinek právě na Vás?
  </li>
</ul>
<p>
  Využijte možnosti individuální lekce.
</p>
<p>
  Po celou dobu 
  <strong>
    individuální lekce
  </strong>
   se věnuji pouze Vám – Vaší kondici, zdravotnímu stavu i přáním přizpůsobuji nejen dlouhodobý cvičební program, ale také každou jednotlivou hodinu.
</p>
<h3>
  Další výhody individuální lekce:
</h3>
<ul>
  <li>
    Cvičební program je sestaven na základě zdravotního stavu, potřeb a přání klienta.
  </li>
  <li>
    Termín konání lekce vychází především z přání klienta.
  </li>
  <li>
    Pro objektivní posouzení výsledků je možné na přání klienta vést fotodokumentaci.
  </li>
  <li>
    Využití cvičební pomůcek:
    <ul>
      <li>
        bosu
      </li>
      <li>
        fitball
      </li>
      <li>
        overball
      </li>
      <li>
        magic circle
      </li>
      <li>
        thera-band
      </li>
      <li>
        core band
      </li>
      <li>
        pilates towel
      </li>
    </ul>
  </li>
</ul>
<h3>
  Co je PILATES?
</h3>
<p>
  Cvičební systém, který rovnoměrně posiluje tělo a zlepšuje jeho držení. Důraz je přitom kladen na správné dýchání, koncentraci, kontrolu pohybů, přesnost a plynulost cviků.
</p>
<p>
  Cviky se zaměřují na posílení tzv. silného centra (oblast beder, břicha od pupku dolů a hýždí), ze kterého všechny pohyby vycházejí.
</p>
<p>
  Rehabilitační metoda je doporučována lékaři a je vhodná pro každý věk pro ženy i muže.
</p>
<h3>
  10 důvodů proč cvičit PILATES
</h3>
<ul>
  <li>
    získáte ploché břicho a štíhlý pas
  </li>
  <li>
    zbavíte se tukových polštářků
  </li>
  <li>
    odstraníte bolesti v zádech
  </li>
  <li>
    pořídíte si dlouhé a štíhlé nohy
  </li>
  <li>
    zpevníte si tělo a zlepšíte jeho držení
  </li>
  <li>
    zbavíte se stresu a únavy
  </li>
  <li>
    naučíte se správně dýchat
  </li>
  <li>
    odstraníte svalové disbalance
  </li>
  <li>
    budete lépe spát
  </li>
  <li>
    zlepšíte si kondici
  </li>
</ul>
`;
